pyatv/mrp/__init__.py

```diff
@@ -326,6 +326,68 @@ def artwork_id(self):
         return MrpPlaying(self.psm.playing)
 
 
+class MrpPower(Power):
+    """Implementation of API for retrieving a power state from an Apple TV."""
+
+    def __init__(self, loop, protocol, remote):
+        """Initialize a new MrpPower instance."""
+        super().__init__()
+        self.loop = loop
+        self.protocol = protocol
+        self.remote = remote
+        self._power_state = None
+        self._listener = None
+        self._add_listeners()
+
+    def _add_listeners(self):
+        self.protocol.add_listener(
+            self.update_power_state,
+            protobuf.DEVICE_INFO_MESSAGE)
+        self.protocol.add_listener(
+            self.update_power_state,
+            protobuf.DEVICE_INFO_UPDATE_MESSAGE)
+
+    async def _turn_on(self):
+        await self.remote.menu()
+
+    async def _turn_off(self):
+        await self.remote.home_hold()
+        await self.remote.select()
+
+    async def _get_current_power_state(self):
+        if self._power_state is None:
+            current_power_state = await self.update_power_state(self.protocol.device_info, protobuf.DEVICE_INFO_MESSAGE)
+            return current_power_state
+        else:
+            return self._power_state
+
+    def power_state(self):
+        """Return device power state."""
+        return self._get_current_power_state()
+
+    def turn_on(self):
+        """Turn device on."""
+        return self._turn_on()
```

Comment:
You can inline this method now:

```python
async def turn_on(self):
    """Turn device on."""
    await self.remote.menu()
```

The pattern I've used where I return a coroutine in non-async methods are bad practice, so I will try change them to async as we go along.